app/src/controllers/pages/selectors.js

```diff
@@ -159,3 +167,74 @@ export const urlOrganizationAndProjectSelector = createSelector(
     return activeProject;
   },
 );
+
+export const activeProjectRoleSelector = createSelector(
+  urlProjectSlugSelector,
+  assignedProjectsSelector,
+  (projectSlug, assignedProjects) => {
+    const assignedProject = assignedProjects[projectSlug];
+
+    return assignedProject?.projectRole;
+  },
+);
+
+const activeOrganizationRoleSelector = createSelector(
+  urlOrganizationSlugSelector,
+  assignedOrganizationsSelector,
+  (organizationSlug, assignedOrganizations) => {
+    const assignedOrganization = assignedOrganizations[organizationSlug];
+
+    return assignedOrganization?.organizationRole;
+  },
+);
+
+export const userRolesSelector = createSelector(
+  userAccountRoleSelector,
+  activeOrganizationRoleSelector,
+  activeProjectRoleSelector,
+  (userRole, organizationRole, projectRole) => ({
+    userRole,
+    organizationRole,
+    projectRole,
+  }),
+);
+
+export const createUserAssignedSelector = (projectSlug, organizationSlug) =>
+  createSelector(
```

Comment:
`createSelector` redundant here as you call `createUserAssignedSelector` each time you use it and it calls all the selectors from scratch without any memorization, so please remove `createSelector` usage here.